### PR TITLE
[Improve] Use 4d input in pytorch poc

### PR DESCRIPTION
**Pull request by wangruohui**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

1. Explicitly add batch size dim for inputs and intermediate features
2. Explicitly set `use_origin` flag to switch between continuous batching and original
3. use float16 by default and accelerate init by skipping initializing parameters on cpu
4. add `model.eval()`
5. support llama for transformers >= 4.32

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
6. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
7. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
8. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/pytorch_poc/engine/engine.py [hunk 1]
```diff
@@ -152,9 +153,12 @@ def __init__(self,
                  model_path: str,
                  scheduler_config: SchedulerConfig = None,
                  cache_config: CacheConfig = None) -> None:
-        hf_model = AutoModelForCausalLM.from_pretrained(model_path)
-        self.patched_model = patch(hf_model, ['context']).cuda()
+        with LoadNoInit():
```
Comment (grimoire) [inline lmdeploy/pytorch_poc/engine/engine.py:156]: what's the differents between this and `from accelerate import init_empty_weights`

File: lmdeploy/pytorch_poc/engine/engine.py [hunk 2]
```diff
@@ -152,9 +153,12 @@ def __init__(self,
...
+            hf_model = AutoModelForCausalLM.from_pretrained(
+                model_path, torch_dtype=torch.float16)
```
Comment (grimoire) [inline lmdeploy/pytorch_poc/engine/engine.py:158]: torch_dtype can be read from `AutoConfig.from_pretrained`

Comment (wangruohui) [reply lmdeploy/pytorch_poc/engine/engine.py]: We can set it to 'auto' to make it read from the config https://github.com/huggingface/transformers/blob/26a38a9033d3fc247aa0c3d9e7afb665f73ec5d5/src/transformers/modeling_utils.py#L2821
